python/private/local_runtime_repo_setup.bzl

```diff
@@ -81,13 +84,15 @@ def define_local_runtime_toolchain_impl(
             # A Python install may not have C headers
             allow_empty = True,
         ),
+        deps = import_deps,
         includes = ["include"],
     )
 
     cc_library(
         name = "_libpython",
         hdrs = [":_python_headers"],
-        deps = _libpython_deps,
+        srcs = [shared_library] if shared_library else [],
```

Comment:
FWIW, I think srcs can be a mix of static and shared libraries. cc_library will sort them out.